core/src/avm2/globals/string.rs

```diff
@@ -356,17 +356,26 @@ fn replace<'gc>(
     let pattern = args.get(0).unwrap_or(&Value::Undefined);
     let replacement = args.get(1).unwrap_or(&Value::Undefined);
     // Handles regex patterns.
-    if let Some(mut regexp) = pattern
+    // doing this a bit awkwardly to avoid holding a lock on the underlying
+    // regexp (if it exists)...
+    let is_regexp = pattern
```

Comment:
I think there's one slightly cleaner (less unwraps) way to write it:
```rs
let regexp = pattern.as_object().and_then(|o| o.as_regexp_object());

if let Some(regexp) = regexp {
    // ...
```

Here, `regexp` becomes `RegExpObject`, so you can pass it to `RegExp::replace_*` methods already type-checked instead of generic `Object`.

Reply:
Changed this up to do as such!